src/kappe/convert.py

```diff
@@ -362,8 +376,24 @@ def process_message(self, msg: WrappedDecodedMessage) -> None:  # noqa: PLR0912
         if self.config.tf_static.remove_tf_static and topic == '/tf_static':
             return
 
-        if topic in ['/tf', '/tf_static']:
-            if not tf_remove(self.config.tf_static, msg):
+        if topic == '/tf':
+            # Apply offsets first if we have any offset configuration but no remove configuration
+            if self.config.tf.offset and not self.config.tf.remove:
+                tf_apply_offset(self.config.tf, msg)
+            # Then handle removal if configured (which also applies offsets internally)
```

Comment:
I wonder if we should not `tf_apply_offset` and do it explicitly here? Problem with offset is you can apply it multiple times.